core/src/test/java/org/elasticsearch/search/aggregations/metrics/stats/multifield/MultiFieldStatsIT.java

```diff
@@ -0,0 +1,380 @@
+    }
+
+    @Override
+    public void testScriptMultiValued() throws Exception {
+        HashMap<String, Script> scripts = new HashMap<>();
+        scripts.put("extract_field", new Script("values", ScriptType.INLINE, AvgIT.ExtractFieldScriptEngine.NAME, Collections.EMPTY_MAP));
+        SearchResponse response = client().prepareSearch("idx").setQuery(matchAllQuery()).addAggregation(multifieldStats(aggName)
+            .fields(Arrays.asList(singleValFields)).scripts(scripts)).execute().actionGet();
+        assertSearchResponse(response);
+        assertHitCount(response, numDocs);
+        assertExpectedStatsResults(response.getAggregations().get(aggName));
+    }
+
+    @Override
+    public void testScriptMultiValuedWithParams() throws Exception {
+        Map<String, Object> params = Collections.singletonMap("s", "values");
+        Map<String, Script> scripts = Collections.singletonMap("field_value",
+            new Script("", ScriptType.INLINE, ValueCountIT.FieldValueScriptEngine.NAME, params));
+        SearchResponse response = client().prepareSearch("idx").setQuery(matchAllQuery()).addAggregation(multifieldStats(aggName)
+            .fields(Arrays.asList(valFieldNames)).scripts(scripts)).execute().actionGet();
+        assertSearchResponse(response);
+        assertHitCount(response, numDocs);
+        assertExpectedStatsResults(response.getAggregations().get(aggName), valFieldNames);
+    }
+
+    private void assertEmptyMultiFieldStats(MultiFieldStats multiFieldStats) {
+        assertThat(multiFieldStats, notNullValue());
+        assertThat(multiFieldStats.getName(), equalTo(aggName));
+        assertThat(multiFieldStats.getFieldCount("value"), equalTo(0L));
+        assertThat(Double.isNaN(multiFieldStats.getMean("value")), is(true));
+        assertThat(Double.isNaN(multiFieldStats.getVariance("value")), is(true));
+        assertThat(Double.isNaN(multiFieldStats.getSkewness("value")), is(true));
+        assertThat(Double.isNaN(multiFieldStats.getKurtosis("value")), is(true));
+        assertThat(Double.isNaN(multiFieldStats.getCovariance("value", "randVal1")), is(true));
+        assertThat(Double.isNaN(multiFieldStats.getCorrelation("value", "randVal1")), is(true));
+    }
+
+    private void assertExpectedStatsResults(MultiFieldStats multiFieldStats) {
+        assertExpectedStatsResults(multiFieldStats, singleValFields);
+    }
+
+    private void assertExpectedStatsResults(MultiFieldStats multiFieldStats, String[] fieldNames) {
+        assertThat(multiFieldStats, notNullValue());
+        assertThat(multiFieldStats.getName(), equalTo(aggName));
+        for (String fieldName : fieldNames) {
+            // check fieldCount
+            assertThat(multiFieldStats.getFieldCount(fieldName), equalTo(multiFieldStatsResults.counts.get(fieldName)));
+            // check mean
+            assertThat(multiFieldStats.getMean(fieldName), closeTo(multiFieldStatsResults.means.get(fieldName), TOLERANCE));
```

Comment:
Consider `org.elasticsearch.test.hamcrest.DoubleMatcher.nearlyEqual()` instead of `closeTo()`.  Absolute tolerances tend to break when the values are very large (returns false due to floating point rounding error) or when the values are very small (returns true, since the absolute difference between two very small numbers is also very small)

https://github.com/elastic/elasticsearch/blob/master/core/src/test/java/org/elasticsearch/test/hamcrest/DoubleMatcher.java